### ISSUE TITLE
hydra-zen not calling os.chdir

### ISSUE THREAD

**Issue by DubiousCactus** [mit-ll-responsible-ai/hydra-zen#394]
Hi,

I'm already quite familiar with hydra and just started using hydra-zen for my new project. I am struggling a little and I think the examples lack consistency (why do I have to do things differently in the [pytorch example](https://mit-ll-responsible-ai.github.io/hydra-zen/how_to/pytorch_lightning.html) than in the tutorials?).
Anyway, when I save files and create directories, they appear at the root of my project and not in the `outputs/` tree as they would when using hydra.
Here is the entry point of my program:
```python

def launch_experiment(exp: BaseExperiementConfig):
    hydra_zen.save_as_yaml(exp, "experiment.yaml")
    "============ CUDA ============"
    model: torch.nn.Module = to_cuda(exp.model)  # type: ignore

    "============ Weights & Biases ============"
    if project_conf.USE_WANDB:
        wandb.init(
            project=project_conf.PROJECT_NAME,
            # config=store(exp),
            # reinit=True, What does this do?
        )

    " ============ Partials instantiation ============ "
    opt = exp.opt(model.parameters())
    train_dataset = exp.train_dataset(tiny=exp.tiny_dataset)
    val_dataset = exp.val_dataset(tiny=exp.tiny_dataset)
    train_batch_sampler = exp.train_batch_sampler(train_dataset)
    val_batch_sampler = exp.val_batch_sampler(val_dataset)

    " ============ Reproducibility of data loaders ============ "
    g = None
    if project_conf.REPRODUCIBLE:
        g = torch.Generator()
        g.manual_seed(0)

    train_loader = exp.train_loader(
        train_dataset, batch_sampler=train_batch_sampler, generator=g
    )
    val_loader = exp.val_loader(
        val_dataset, batch_sampler=val_batch_sampler, generator=g
    )

    " ============ Training ============ "
    BaseTrainer(
        model=model,
        opt=opt,
        train_loader=train_loader,
        val_loader=val_loader,
    ).train(
        epochs=exp.epochs,
        val_every=exp.val_every,
        visualize_every=exp.visualize_every,
    )


if __name__ == "__main__":
    " ============ Hydra-Zen ============ "
    store.add_to_hydra_store()
    zen(
        launch_experiment,
        pre_call=hydra_zen.zen(
            lambda exp: seed_everything(exp.seed) if project_conf.REPRODUCIBLE else lambda: None
        ),
    ).hydra_main(
        config_name="exp/lnp",
        version_base="1.2",  # Hydra base version
    )
```

In my class `BaseTrainer`, when I save whatever file, it's written in the root directory and not in my run folder in `outputs`.
Am I missing something here? 

Thank you!

**Comment by DubiousCactus** [comment]
Aah I found the issue! I set `version_base="1.2"`, but it turns out that [hydra disabled this feature](https://hydra.cc/docs/1.2/upgrades/1.1_to_1.2/changes_to_job_working_dir/) from that version.

**Comment by jgbos** [comment]
Interestingly I was hitting this exact same error yesterday as I meant to use "1.1" instead of "1.2"!

**Comment by DubiousCactus** [comment]
I would still like this feature for the latest version though. How can I set `hydra.job.chdir=True`?

**Comment by jgbos** [comment]
Hmm...  I really need to spend more time playing with configuring Hydra via dataclasses.  Here is a solution I came up with:

```python
import os

from hydra_zen import zen, store, make_config
from hydra.conf import HydraConf, JobConf


store(
    make_config(job=JobConf(chdir=True), bases=(HydraConf,)),
    name="config",
    group="hydra",
)

store(make_config(), name="config")


def task():
    print(os.getcwd())


if __name__ == "__main__":
    store.add_to_hydra_store(overwrite_ok=True)
    zen(task).hydra_main(config_path=None, config_name="config", version_base="1.2")

```

Let me know if you have issues with this solution though.

**Comment by rsokl** [comment]
Hi @DubiousCactus , thanks for taking time to write this up, and I'm glad you found the solution! 

> I am struggling a little and I think the examples lack consistency (why do I have to do things differently in the [pytorch example](https://mit-ll-responsible-ai.github.io/hydra-zen/how_to/pytorch_lightning.html) than in the tutorials?)

We are always eager to improve the quality of our examples/documentation. In what way did the pytorch example differ from the tutorials? Any other insights you have into how we might improve our docs/consistency between examples would be much appreciated!

**Comment by rsokl** [comment]
@jgbos  here is a slightly simplified version, replace

```python
store(
    make_config(job=JobConf(chdir=True), bases=(HydraConf,)),
    name="config",
    group="hydra",
)
```

with

```python
store(HydraConf(job=JobConf(chdir=True)), name="config", group="hydra")
```

so in total this looks like:

```python
import os

from hydra_zen import zen, store, make_config
from hydra.conf import HydraConf, JobConf


store(HydraConf(job=JobConf(chdir=True)), name="config", group="hydra")

store(make_config(), name="config")


def task():
    print(os.getcwd())


if __name__ == "__main__":
    store.add_to_hydra_store(overwrite_ok=True)
    zen(task).hydra_main(config_path=None, config_name="config", version_base="1.2")
```

this would be a nice, short How-To for us to document

**Comment by jgbos** [comment]
@rsokl yeah, I should have used that one since we just overwrite the existing config.  I was in the the mindset of creating multiple configs.

I also agree that this example and #389 should be added to the docs.  I'll try to get something up soon.  I will also take a look at generating a simple pytorch example without using lightning.

**Comment by jgbos** [comment]
As for the `overwite_ok` flag, here is an example that utilizes two stores, one for `hydra` and one for the experiment:

```python
import os

from hydra_zen import zen, store, make_config, ZenStore
from hydra.conf import HydraConf, JobConf

hydra_store = ZenStore(overwrite_ok=True)

hydra_store(
    HydraConf(job=JobConf(chdir=True)),
    name="config",
    group="hydra",
)

store(make_config(), name="config")


def task():
    print(os.getcwd())


if __name__ == "__main__":
    hydra_store.add_to_hydra_store()
    store.add_to_hydra_store()
    zen(task).hydra_main(config_path=None, config_name="config", version_base="1.2")
```

**Comment by DubiousCactus** [comment]
> 

Hi @rsokl, thanks for the quick reply!

To be honest I found it really hard to understand how to use hydra-zen, and mostly the advantage over Hydra. To me, Hydra is great because I can extract the config from my code and isolate it in simple YAML files which are easy to find and read, so it gives me a clear view of my experiments. Being able to create a hierarchy and components is also a huge advantage: if I want to change dataset for my experiment, I can use one dataset configuration. With hydra-zen, I'm not sure how to do that because I can't find clear examples in the docs. It's shown in the [configuration groups](https://mit-ll-responsible-ai.github.io/hydra-zen/tutorials/config_groups.html), but I can't make it work because I don't understand how groups nesting works and so I can't understand the error.

I'm starting to see the point now, which is probably mostly advantageous to non-experiment-based apps because you don't want to write YAML configs to keep the code DRY. But  for experiment-based things like ML, I want to be able to specify various experiments easily and also use the CLI to play with different config components. The [boiler-plate free ML experiment example](https://mit-ll-responsible-ai.github.io/hydra-zen/how_to/pytorch_lightning.html) briefly shows how to do this but it's not complete enough.

Regarding the inconsistency, most examples use the `@store` decorator with the `zen.hydra_main()` function and `config_name="my_app"`. It's not clear at all why I need this extra config `my_app` on top. But then, in the ML example, the decorator is removed and we now have to use ` store = ZenStore(deferred_hydra_store=False)` for reasons I had to figure out from API reference. And if I have a base experiment and want specialized versions of that, how do I pass those to `train_and_eval()`?

I actually enjoy using this library, but it's still quite blurry in some places and that leads to frustration when I want to do the same things as I used to do easily with hydra. I think the docs and the tutorials could be greatly improved by at least explaining what the main things do (`@store` decorator, `ZenStore(deferred_hydra_store=False)`, etc.) and why we use them. It also took me 2 days to figure out how to make composable configurations with a common base, and that I needed partials to specialize a base config (unless there's an easier way with DataClasses?).

Thanks!

**Comment by rsokl** [comment]
@DubiousCactus we have [added a few How-To sections to our documentation](https://mit-ll-responsible-ai.github.io/hydra-zen/changes.html#documentation-2023-01-22) including a section about configuring Hydra's chdir behavior.

We will also reflect on the rest of the feedback that you have provided, and plan to improve our tutorials and explanations. Thank you for taking the time to share your experience with us, and apologies for any frustration you experienced.
